wrapper.go

```diff
@@ -28,6 +29,48 @@ const (
 	HSMAutoDeprecated = "hsm-auto"
 )
 
+// KeyType defines types of cryptographic keys.
+type KeyType string
+
+const (
+	RSA2048 KeyType = "rsa_2048"
```

Comment:
Any reason why we're going with `rsa_2048` for the string value representation (as opposed to `rsa-2048` or `rsa2048`)? Keep in mind that these are shared across provider implementations, so what's likely to work for one provider may not for work for another if these are intended to be passed as string values to the provider's SDK APIs. Same for purpose and protection level.

Reply:
There's no particular reason the string value is `rsa_2048` for `KeyType`. These constants are meant to be generic and work across different providers. The intent is for the implementations of `LifecycleWrapper` methods to map the 
 `KeyType` to a value that's expected by the specific provider's SDK APIs. Taking Azure Key Vault as an example, the `RSA2048` `KeyType` would map to one of [these values](https://github.com/Azure/azure-sdk-for-go/blob/master/services/keyvault/v7.0/keyvault/models.go#L173). The same is true for purpose and protection level.